src/sinks/datadog/logs/tests.rs

```diff
@@ -533,3 +535,96 @@ async fn error_is_retriable() {
     //       but are not straightforward to instantiate due to the design of
     //       the crates they originate from.
 }
+
+#[tokio::test]
```

Comment:
Nice tests! 😄 

While redundant, can we add the same tests to the Datadog metrics sink to ensure that there aren't any regressions there either?